delivery/repositories/runfolder_repository.py

```diff
@@ -24,6 +24,29 @@ def __init__(self, base_path, file_system_service=FileSystemService()):
         self._base_path = base_path
         self.file_system_service = file_system_service
 
+    def _add_projects_to_runfolder(self, runfolder):
+        try:
+            projects_base_dir = os.path.join(runfolder.path, "Projects")
+            project_directories = self.file_system_service.find_project_directories(
+                projects_base_dir)
+
+            def project_from_dir(d):
+                return RunfolderProject(
+                    name=os.path.basename(d),
+                    path=os.path.join(projects_base_dir, d),
+                    runfolder_path=runfolder.path)
+
+            # There are scenarios where there are no project directories in the runfolder,
+            # i.e. when fastq files have not yet been divided into projects
+            if project_directories:
+                runfolder.projects = list(map(
+                    project_from_dir, project_directories))
+        except FileNotFoundError as e:
+            log.warning("Did not find Project folder for: {}".format(runfolder.name))
+            pass
+        finally:
+            return runfolder
```

Comment:
I guess you could remove this return statement, as you're not catching it anyway, and as you only seem to be changing the list property of the object (mutable) anyway. Would make the code a bit more clear.